### PR TITLE
SPI collect W && UHCI::REG_Q regs

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by MabezDev** [comment]
Sorry for the delay @burrbull, much of our team was busy the last couple of weeks. We're about to cut a release of esp-hal etc, after that I will review your PAC PRs

**Comment by burrbull** [comment]
rebased

File: esp32/src/spi0.rs [hunk 1]
```diff
@@ -205,85 +190,16 @@ impl RegisterBlock {
     pub const fn slv_rd_bit(&self) -> &SLV_RD_BIT {
         &self.slv_rd_bit
     }
-    #[doc = "0x80 - "]
+    #[doc = "0x80..0xc0 - "]
     #[inline(always)]
-    pub const fn w0(&self) -> &W0 {
-        &self.w0
+    pub const fn w(&self, n: usize) -> &W {
```
Comment (MabezDev) [inline esp32/src/spi0.rs:195]: @burrbull is there a way in svd2rust to treat a cluster of registers as slice? In the case of the W register here and many others, we often cast the first to a pointer and create a slice from that, maybe svd2rust can do this for us?

**Review by MabezDev — APPROVED** [review]
Sorry again for the delay! These changes look good to me, thank you!

File: esp32/src/spi0.rs (hunk 1 above)
Comment (burrbull) [reply esp32/src/spi0.rs:195]: P.S. Looks like previous example could use iterator.

**Pull request merged**